Horsemanship

TheThanathor opened the pull request:
Level up your horses by using them to improve their health, speed and jump strength

Full module description will be added soon once the design is more finalized

TheThanathor committed:
update old module to 1.21.5

TheThanathor committed:
add swimming with rider

TheThanathor committed:
fix horse breeding, add varying scales

TheThanathor committed:
update horse training mechanics

TheThanathor committed:
add more sounds and particles

TheThanathor committed:
swap carrot to standard foods

TheThanathor committed:
update and simplify Horsemanship

TheThanathor committed:
update headers and reorganize folders

TheThanathor committed:
update horse flight

TheThanathor commented:
Ready for review, but needs textures and the pack.png before merging

runcows commented on gm4_horsemanship/data/gm4_horsemanship/recipe/horse_wings.json:
```diff
@@ -0,0 +1,80 @@
+            "minecraft:golden_horse_armor",
+            "minecraft:iron_horse_armor",
+            "minecraft:leather_horse_armor"
+        ]
```
This should probably be defined in an item tag. (Thinking about copper and netherite horse armor)

runcows commented on gm4_horsemanship/README.md:
```diff
@@ -0,0 +1,9 @@
+
+### Features
+- Light portals of any shape with Flint and Steel or Fire Charges.
+- Portals can use Crying Obsidian.
```
Needs updated

runcows commented on gm4_horsemanship/data/gm4_horsemanship/advancement/feed_horse.json:
```diff
@@ -0,0 +1,187 @@
+    "apple": {
+      "trigger": "minecraft:player_interacted_with_entity",
+      "conditions": {
+        "player": [],
```
I believe the player fields in all of these criteria can be unset. Applies to other advancements too

runcows commented on gm4_horsemanship/assets/translations.csv:
```diff
@@ -0,0 +1,2 @@
+key,en_us
+gm4.horsemanship.item.horse_wings,Horse Wings
```
Related because I don't know where else to put it. A Guidebook entry is missing

runcows commented on gm4_horsemanship/data/gm4_horsemanship/function/horse_processing/food/check_horse.mcfunction:
```diff
@@ -0,0 +1,9 @@
+# run from horse_processing/food/feed_original_food
+
+tag @s add gm4_horse_check
+execute as @p[tag=gm4_horse_check_player] if predicate gm4_horsemanship:check_horse_looking_at run tag @e[type=#gm4_horsemanship:trainable,tag=gm4_horse_check,distance=..8,limit=1] add gm4_horse_target
```
The `@e` can be avoided by setting a scoreboard value and checking it in the following line to tag `@s`

runcows commented on gm4_horsemanship/data/gm4_horsemanship/function/horse_processing/food/type/chorus_fruit.mcfunction:
```diff
@@ -0,0 +1,32 @@
+# run from horse_processing/food/feed_added_food_untamed
+
+# TODO
+say vwoop!
```
todo? 👀

runcows commented on gm4_horsemanship/data/gm4_horsemanship/function/horse_processing/food/feed_original_food.mcfunction:
```diff
@@ -0,0 +1,21 @@
+execute if score $food_processed gm4_horse_data matches 0 if entity @s[advancements={gm4_horsemanship:feed_horse={golden_carrot=true}}] as @e[type=#gm4_horsemanship:trainable,tag=gm4_horse_target,distance=..8,limit=1] at @s run function gm4_horsemanship:horse_processing/food/type/golden_carrot
+execute if score $food_processed gm4_horse_data matches 0 if entity @s[advancements={gm4_horsemanship:feed_horse={sugar=true}}] as @e[type=#gm4_horsemanship:trainable,tag=gm4_horse_target,distance=..8,limit=1] at @s run function gm4_horsemanship:horse_processing/food/type/sugar
+execute if score $food_processed gm4_horse_data matches 0 if entity @s[advancements={gm4_horsemanship:feed_horse={wheat=true}}] as @e[type=#gm4_horsemanship:trainable,tag=gm4_horse_target,distance=..8,limit=1] at @s run function gm4_horsemanship:horse_processing/food/type/wheat
+execute if score $food_processed gm4_horse_data matches 0 if entity @s[advancements={gm4_horsemanship:feed_horse={hay_block=true}}] as @e[type=#gm4_horsemanship:trainable,tag=gm4_horse_target,distance=..8,limit=1] at @s run function gm4_horsemanship:horse_processing/food/type/hay_block
```
I think most, if not all, of the `food/type/` functions dont need to be run at the horse.

runcows reviewed (commented):
Partial Review. I didn't get through a lot of the functions, but I figured something is better than nothing :D

TheThanathor committed:
quick wing period fix

TheThanathor committed:
runcow's horse improvements

runhorse if you will

TheThanathor committed:
add horse wing texture (WIP)

TheThanathor committed:
fix the wings!

TheThanathor committed:
oops missing tag

TheThanathor committed:
add advancement, image and website note

TheThanathor committed:
update to 1.21.11

TheThanathor committed:
add translations

TheThanathor committed:
technically RP is not required rn

misode committed:
Backport mob_griefing gamerule

runcows commented on gm4_guidebook/triggers.json:
```diff
@@ -50,6 +50,7 @@
   "heart_canisters": 15,
   "helious_shamir": 69,
   "holographic_tags": 94,
+  "horsemanship": 122,
```
122 is taken by lavish livestock right now

misode reviewed (commented):
I've just had a go testing this module in-game. I don't think this is mergeable right now.

* There doesn't seem to be any documentation of its features (wiki, readme, guidebook) meaning you have to read the code in order to know what to feed your horse.
* From some quick survival testing, it doesn't seem very fun? The initial speed of your horse is reduced a ton meaning you have to spend a lot of time just slowly riding your horse before it becomes useful.
* The level and stamina systems are not very visible, I had to cheat and look at the scoreboards to make any sense of it.

TheThanathor committed:
update Horsemanship

- removed Stamina mechanics
- updated Guidebook entries
- level 0 horses have higher stats (level 15 horses remain the same)

TheThanathor committed:
update guidebook trigger

TheThanathor committed:
fix horse jumping

TheThanathor committed:
fix attribuite names

TheThanathor committed:
fix guidebook trigger, remove plugin change

Bloo-dev commented on gm4_horsemanship/data/gm4_horsemanship/function/horse_processing/glider/wing/init.mcfunction:
```diff
@@ -0,0 +1,17 @@
+
```
Missing header comment

Bloo-dev commented on gm4_horsemanship/data/gm4_horsemanship/function/horse_processing/glider/wing/calculate_wing_pitch.mcfunction:
```diff
@@ -0,0 +1,37 @@
+
```
Missing header comment

Bloo-dev commented on gm4_horsemanship/beet.yaml:
```diff
@@ -0,0 +1,31 @@
+      description: Form a bond with your horses
+      recommended: []
+      notes:
+        - Resourcepack is recommended with this module.
```
```suggestion
        - Usage of the Gamemode 4 Resource Pack is highly recommended.
```

Bloo-dev reviewed (changes requested):
Full review. I mostly found some minor format issues.

TheThanathor committed:
resolve bloo's comments

TheThanathor committed:
fix last missing attribute

Bloo-dev commented:
With #1276 merged this no longer errors. (once rebased)

runcows commented on gm4_horsemanship/data/gm4_horsemanship/recipe/horse_wings.json:
```diff
@@ -0,0 +1,76 @@
+                "shearing_sound": "minecraft:item.horse_armor.unequip",
+                "slot": "body"
+            },
+            "minecraft:custom_model_data": "item/horse_wings",
```
Since this PR has been retargeted at 26.1, this is no longer valid.

```suggestion
            "minecraft:custom_model_data": {"strings":["gm4_horsemanship:item/horse_wings"]},
```

runcows commented on gm4_horsemanship/data/gm4/advancement/horse_takes_flight.json:
```diff
@@ -0,0 +1,32 @@
+        "icon": {
+            "id": "minecraft:elytra",
+            "components": {
+                "minecraft:custom_model_data": "item/horse_wings"
```
Same issue here

TheThanathor committed:
Merge remote-tracking branch 'upstream/master' into horsemanship

TheThanathor committed:
update cmd for 26.1

misode commented on gm4_horsemanship/data/gm4_horsemanship/function/tick.mcfunction:
```diff
@@ -0,0 +1,5 @@
+schedule function gm4_horsemanship:tick 1t
+
+# glider
+execute as @e[type=#gm4_horsemanship:trainable,tag=gm4_horse.gliding] on passengers if entity @s[type=item_display,tag=gm4_horse.glider_wing] run function gm4_horsemanship:horse_processing/glider/wing/update
```
If the horse is already gliding, why check for the existence of the display entity? Is that ever not present?

TheThanathor replied:
it's executing on all passengers, but that includes the player, so I filter for just the display entity

misode commented on gm4_horsemanship/data/gm4_horsemanship/gm4_recipes/horse_wings.json:
```diff
@@ -0,0 +1,20 @@
+{
```
This file is no longer necessary. The guidebook now reads the recipe itself

misode commented on gm4_horsemanship/data/gm4_horsemanship/recipe/horse_wings.json:
```diff
@@ -0,0 +1,80 @@
+{
```
Add an advancement that gives this recipe when you have an elytra. See the vanilla `recipes/...` advancements as reference.

misode commented on gm4_horsemanship/data/gm4_horsemanship/tags/item/horse_armor.json:
```diff
@@ -0,0 +1,10 @@
+    "minecraft:iron_horse_armor",
+    "minecraft:golden_horse_armor",
+    "minecraft:diamond_horse_armor",
+    {"id":"minecraft:netherite_horse_armor","required":false}
```
These can now all be required

misode commented on gm4_horsemanship/data/gm4_horsemanship/function/horse_processing/glider/wing/update.mcfunction:
```diff
@@ -0,0 +1,11 @@
+# at unspecified
+# run from tick
+
+execute on vehicle store result score $set_yaw gm4_horse_data run data get entity @s Rotation[0] 100
```
Profiling the datapack when a couple players are using the horse with wings, this is a very slow command (1mspt). But I assume it needs to run every tick to make the animation look good?

<img width="2456" height="509" alt="Image" src="https://github.com/user-attachments/assets/0afe1546-b4c9-4b55-8998-0da178f8cd39" />

TheThanathor committed:
update!

added recipe advancement
fixed horses losing wings mid flight
added feather falling support
fixed guidebook?
horses now need to be moving to level up

TheThanathor committed:
Merge remote-tracking branch 'upstream/master' into horsemanship

misode reviewed (approved):
Review changed were addressed. I did a final in-game test and things are working.
